test/prefer-string-replace-all.mjs

```diff
@@ -14,6 +14,7 @@ test({
 		'foo.replace(/[a]/g, bar)',
 		'foo.replace(/a?/g, bar)',
 		'foo.replace(/.*/g, bar)',
+		'foo.replace(/a|b/g, bar)',
```

Comment:
Can you add a test for escaped `|`?

Reply:
Should be resolved in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1962/commits/fbbccddcec24f9d03b2330ba6ea4138f0738f1bc, and the `unicorn/better-regex` error was solved in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1962/commits/f67a9f2ca208be1763d27a2af09b0dfcf39ba279.